uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -680,4 +724,753 @@ public function testCalculateBalanceChangeForALeaveRequestForAContactWithMultipl
+  }
+
+  public function testALeaveRequestWhenThereAreNoOverlappingLeaveRequests() {
+    $contactID = 1;
+    $publicHoliday = new PublicHoliday();
+    $publicHoliday->date = '2016-11-11';
+
+    $fromDate1 = new DateTime('2016-11-02');
+    $toDate1 = new DateTime('2016-11-04');
+
+    $fromDate2 = new DateTime('2016-11-05');
+    $toDate2 = new DateTime('2016-11-10');
+
+    $period = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      'end_date'   => CRM_Utils_Date::processDate('2016-12-31'),
+    ]);
+
+    $periodEntitlement = LeavePeriodEntitlementFabricator::fabricate([
+      'type_id' => $this->absenceType->id,
+      'contact_id' => $contactID,
+      'period_id' => $period->id
+    ]);
+
+    $this->createLeaveBalanceChange($periodEntitlement->id, 20);
+
+    $workPattern = WorkPatternFabricator::fabricateWithA40HourWorkWeek();
+    ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $contactID,
+      'pattern_id' => $workPattern->id
+    ]);
+
+    $leaveRequestStatuses = array_flip(LeaveRequest::buildOptions('status_id'));
+    $leaveRequest1 = LeaveRequestFabricator::fabricateWithoutValidation([
+      'type_id' => $this->absenceType->id,
+      'contact_id' => $contactID,
+      'status_id' => $leaveRequestStatuses['Waiting Approval'],
+      'from_date' => $fromDate1->format('YmdHis'),
+      'from_date_type' => 1,
+      'to_date' => $toDate1->format('YmdHis'),
+      'to_date_type' => 1
+    ], true);
+
+    $leaveRequest2 = LeaveRequestFabricator::fabricateWithoutValidation([
+      'type_id' => $this->absenceType->id,
+      'contact_id' => $contactID,
+      'status_id' => $leaveRequestStatuses['Rejected'],
+      'from_date' => $fromDate2->format('YmdHis'),
+      'from_date_type' => 1,
+      'to_date' => $toDate2->format('YmdHis'),
+      'to_date_type' => 1
+    ], true);
+
+    PublicHolidayLeaveRequestFabricator::fabricate($contactID, $publicHoliday);
+    $publicHolidayLeaveRequest = LeaveRequest::findPublicHolidayLeaveRequest($contactID, $publicHoliday);
+
+    //this date overlapps with public holiday and a Rejected status leave request
+    $fromDate = new DateTime('2016-11-05');
+    $toDate = new DateTime('2016-11-11');
+    $leaveRequest = LeaveRequest::create([
+      'type_id' => $this->absenceType->id,
+      'contact_id' => 1,
+      'status_id' => 1,
+      'from_date' => $fromDate->format('YmdHis'),
+      'from_date_type' => 1,
+      'to_date' => $toDate->format('YmdHis'),
+      'to_date_type' => 1
+    ]);
+    $this->assertEquals($leaveRequest->from_date, $fromDate->format('YmdHis'));
+  }
+
+  /**
+   * @expectedException CRM_HRLeaveAndAbsences_Exception_InvalidLeaveRequestException
+   * @expectedExceptionMessage Balance change for the leave request cannot be greater than the remaining balance of the period
+   */
+  public function testLeaveRequestCannotBeCreatedWhenBalanceChangeGreaterThanPeriodEntitlementBalanceChange() {
```

Comment:
Please also amend the test title to include the allow overuse information. Otherwise, it looks like the validation is independent of the rules on the absence type